src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -332,14 +332,24 @@ private static int VerifyClientCertificate(int preverify_ok, IntPtr x509_ctx_ptr
 
         private static unsafe int AlpnServerSelectCallback(IntPtr ssl, out IntPtr outp, out byte outlen, IntPtr inp, uint inlen, IntPtr arg)
         {
+            outp = IntPtr.Zero;
+            outlen = 0;
+
+            if (arg == IntPtr.Zero)
+            {
+                return Interop.Ssl.SSL_TLSEXT_ERR_NOACK;
+            }
+
             GCHandle protocols = GCHandle.FromIntPtr(arg);
+            if (!protocols.IsAllocated || protocols.Target == null)
```

Comment:
In what situation would it be freed?  If there's a race condition where it could be freed before this callback, is it possible it could be freed during?

Reply:
Well not anymore, before there was, I could remove that code in theory. I actually put it in, in my first round of trying to find the issue, thought it was worth leaving in for now. A Debug.Assert might suffice, which was my original idea. 

https://github.com/dotnet/corefx/pull/24389#discussion_r142327559